### PR TITLE
search: Improve no results message in repo section

### PR DESCRIPTION
**Pull request by fkling**
I added a new prop to the SidebarSection component that allows for customizing the "No result" message.

### CODE REVIEW

**Review by rrhyne — APPROVED** [review]
Looks great, thanks for taking this one on!

File: client/web/src/search/results/sidebar/SearchSidebar.tsx [hunk 1]
```diff
@@ -96,6 +96,8 @@ export const SearchSidebar: React.FunctionComponent<SearchSidebarProps> = props
         [persistToggleState, props.telemetryService]
     )
 
+    const repoFilterLinks = getRepoFilterLinks(props.filters, onDynamicFilterClicked)
```
Comment (limitedmage) [inline client/web/src/search/results/sidebar/SearchSidebar.tsx:99]: Should this be in a `useMemo` maybe?

File: client/web/src/search/results/sidebar/SearchSidebarSection.tsx [hunk 2]
```diff
@@ -14,7 +14,11 @@ export const SearchSidebarSection: React.FunctionComponent<{
     showSearch?: boolean // Search only works if children are FilterLink
     onToggle?: (open: boolean) => void
     open?: boolean
-}> = ({ header, children = [], className, showSearch = false, onToggle, open }) => {
+    /**
+     * Shown when the built-in search doesn't find any results.
+     */
+    noResultText?: React.ReactElement | string
+}> = ({ header, children = [], className, showSearch = false, onToggle, open, noResultText = 'No result' }) => {
```
Comment (limitedmage) [inline client/web/src/search/results/sidebar/SearchSidebarSection.tsx:21]:
```suggestion
}> = ({ header, children = [], className, showSearch = false, onToggle, open, noResultText = 'No results' }) => {
```

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/results/sidebar/SearchSidebar.tsx (hunk 1 above)
Comment (fkling) [reply client/web/src/search/results/sidebar/SearchSidebar.tsx:99]: Ah yes, good point.

**Pull request merged**